private/bufpkg/bufimage/bufimagebuild/builder.go

```diff
@@ -278,108 +207,63 @@ func getBuildResult(
-		descFileDescriptors = append(descFileDescriptors, buildResult.DescFileDescriptors...)
-	}
-	return checkAndSortDescFileDescriptors(descFileDescriptors, rootRelFilePaths)
-}
-
 // We need to sort the FileDescriptors as they may/probably are out of order
 // relative to input order after concurrent builds. This mimics the output
 // order of protoc.
-func checkAndSortDescFileDescriptors(
-	descFileDescriptors []*desc.FileDescriptor,
+func checkAndSortFileDescriptors(
+	fileDescriptors []protoreflect.FileDescriptor,
 	rootRelFilePaths []string,
-) ([]*desc.FileDescriptor, error) {
-	if len(descFileDescriptors) != len(rootRelFilePaths) {
-		return nil, fmt.Errorf("rootRelFilePath length was %d but FileDescriptor length was %d", len(rootRelFilePaths), len(descFileDescriptors))
-	}
-	nameToDescFileDescriptor := make(map[string]*desc.FileDescriptor, len(descFileDescriptors))
-	for _, descFileDescriptor := range descFileDescriptors {
-		// This is equal to descFileDescriptor.AsFileDescriptorProto().GetName()
-		// but we double-check just in case
-		//
-		// https://github.com/jhump/protoreflect/blob/master/desc/descriptor.go#L82
-		name := descFileDescriptor.GetName()
+) ([]protoreflect.FileDescriptor, error) {
+	if len(fileDescriptors) != len(rootRelFilePaths) {
+		return nil, fmt.Errorf("rootRelFilePath length was %d but FileDescriptor length was %d", len(rootRelFilePaths), len(fileDescriptors))
+	}
+	nameToFileDescriptor := make(map[string]protoreflect.FileDescriptor, len(fileDescriptors))
+	for _, fileDescriptor := range fileDescriptors {
+		name := fileDescriptor.Path()
 		if name == "" {
 			return nil, errors.New("no name on FileDescriptor")
 		}
-		if name != descFileDescriptor.AsFileDescriptorProto().GetName() {
-			return nil, errors.New("name not equal on FileDescriptorProto")
-		}
-		if _, ok := nameToDescFileDescriptor[name]; ok {
+		if _, ok := nameToFileDescriptor[name]; ok {
 			return nil, fmt.Errorf("duplicate FileDescriptor: %s", name)
 		}
-		nameToDescFileDescriptor[name] = descFileDescriptor
+		nameToFileDescriptor[name] = fileDescriptor
 	}
 	// We now know that all FileDescriptors had unique names and the number of FileDescriptors
 	// is equal to the number of rootRelFilePaths. We also verified earlier that rootRelFilePaths
 	// has only unique values. Now we can put them in order.
-	sortedDescFileDescriptors := make([]*desc.FileDescriptor, 0, len(descFileDescriptors))
+	sortedFileDescriptors := make([]protoreflect.FileDescriptor, 0, len(fileDescriptors))
 	for _, rootRelFilePath := range rootRelFilePaths {
-		descFileDescriptor, ok := nameToDescFileDescriptor[rootRelFilePath]
+		fileDescriptor, ok := nameToFileDescriptor[rootRelFilePath]
 		if !ok {
 			return nil, fmt.Errorf("no FileDescriptor for rootRelFilePath: %q", rootRelFilePath)
 		}
-		sortedDescFileDescriptors = append(sortedDescFileDescriptors, descFileDescriptor)
-	}
-	return sortedDescFileDescriptors, nil
-}
-
-func getSyntaxUnspecifiedFilenamesFromBuildResults(
```

Comment:
Why where these removed? I'm likely just not looking carefully enough, but `bufimagebuild` needs to produce a full `Image`, which includes these extra things that `FileDescriptorSets` do not have.